src/CMakeLists.txt

```diff
@@ -82,8 +82,6 @@ libcvc4_add_sources(
   preprocessing/passes/non_clausal_simp.h
   preprocessing/passes/pseudo_boolean_processor.cpp
   preprocessing/passes/pseudo_boolean_processor.h
-  preprocessing/passes/quantifier_macros.cpp
-  preprocessing/passes/quantifier_macros.h
```

Comment:
I think @4tXJ7f 's general input about this would be nice. If I remember correctly, quantifier macros was once upon a time in `theory/quantifiers` and then it was moved to `preprocessing`, right? Now we are bringing them back to `theory/quantifiers`. Do we expect other preprocessing passes to have the same fate? And can this recreate the problems that made us move all passes to `preprocessing` in the first place?

Reply:
There are no problems in this PR.  Another way to view this PR is that we are merging quantifier_macros and non_clausal_simplify, the latter subsumes the former.

Reply:
Or another explanation: the original problem was that we were using custom calls to `theory/`. In this PR, we are using an existing interface (ppAssert) which is called from non-clausal simplification.